### ISSUE TITLE
Ciilium 1.11.3 not deploying correctly on k8s with COS - docker runtime

### ISSUE THREAD

**Issue by hoozgomk** [cilium/cilium#19408]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Hello,
I'm trying to deploy Cilium ver 1.11.3 with COS - docker runtime on k8s cluster on GKE.

I'm using helm chart:

          helm upgrade cilium cilium/cilium --version 1.11.3 --install \
              --namespace kube-system \
              --set nodeinit.enabled=true \
              --set nodeinit.reconfigureKubelet=true \
              --set nodeinit.restartPods=true \
              --set nodeinit.removeCbrBridge=true \
              --set cni.binPath=/home/kubernetes/bin \
              --set gke.enabled=true \
              --set ipam.mode=kubernetes \
              --set ipv4NativeRoutingCIDR=$NATIVE_CIDR \
              --set prometheus.enabled=true \
              --set operator.prometheus.enabled=true \
              --set hubble.enabled=true \
              --set hubble.ui.enabled=true \
              --set hubble.relay.enabled=true \
              --set containerRuntime.integration=docker \
              --set hubble.metrics.enabled="{dns,drop,tcp:destinationContext=pod|ip;sourceContext=pod|ip,flow:destinationContext=pod|ip;sourceContext=pod|ip,port-distribution,icmp,http:destinationContext=pod|ip;sourceContext=pod|ip}" \
              --set hubble.metrics.port=9096
              
After the deployment, I see that some pods have problems, for example hubble-relay is not started successfully and have following events:
                                                                 
```
Warning  FailedMount      2m29s                  kubelet            MountVolume.SetUp failed for volume "tls" : failed to sync secret cache: timed out waiting for the condition       
Warning  FailedMount      2m29s                  kubelet            MountVolume.SetUp failed for volume "config" : failed to sync configmap cache: timed out waiting for the condition 
Warning  NetworkNotReady  2m6s (x14 over 2m30s)  kubelet            network is not ready: container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message │:docker: network plugin is not ready: cni config uninitialized                                                                                                                                     
Normal   Created          119s                   kubelet            Created container hubble-relay                       
Normal   Started          119s                   kubelet            Started container hubble-relay
Warning  Unhealthy        101s (x2 over 111s)    kubelet            Liveness probe failed: dial tcp 10.88.2.5:4245: i/o timeout                                                   
Warning  Unhealthy        99s (x3 over 117s)     kubelet            Readiness probe failed: dial tcp 10.88.2.5:4245: i/o timeout
```

Some other pods, like koonectivity-agent, l7-default-backend and others are failing because with similar readiness and liveness probes failures:
```
Warning  Unhealthy         2m8s (x11 over 7m8s)    kubelet            Liveness probe failed: Get "http://10.88.0.4:8080/healthz": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
Warning  Unhealthy  4m30s  kubelet            Readiness probe failed: timeout: failed to connect service ":8080" within 1s     
```

Basically, all cluster becomes not operational.

The exact same setup works correctly when deployed on GKE k8s cluster with containerd runtime, so it seems that this problem is related to k8s cluster runtime.

Kind regards,

### Cilium Version

1.11.3

### Kernel Version

Linux gke 5.4.144+ #1 SMP Wed Nov 3 09:56:10 PDT 2021 x86_64 GNU/Linux

### Kubernetes Version

v1.21.6-gke.1503

### Sysdump

[cilium-sysdump-20220412-124919.zip](https://github.com/cilium/cilium/files/8472213/cilium-sysdump-20220412-124919.zip)

### Anything else?

I also noticed that previous version 1.11.2 was not even deploying on k8s cluster with COS, cluster was becoming not operational, nodes were in NotReady state.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
